x-pack/plugins/index_lifecycle_management/__jest__/client_integration/edit_policy/features/node_allocation/general_behavior.helpers.ts

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { createNodeAllocationActions, savePolicy, setReplicas } from '../../../helpers';
+import { initTestBed } from '../../init_test_bed';
+
+type SetupReturn = ReturnType<typeof setupGeneralNodeAllocation>;
+
+export type GeneralNodeAllocationTestBed = SetupReturn extends Promise<infer U> ? U : SetupReturn;
+
+export const setupGeneralNodeAllocation = async () => {
+  const testBed = await initTestBed();
+
+  return {
+    ...testBed,
+    actions: {
+      ...createNodeAllocationActions(testBed, 'warm'),
+      savePolicy: () => savePolicy(testBed),
+      setReplicas: (value: string) => setReplicas(testBed, 'warm', value),
+    },
```

Comment:
I agree the naming can still be improved. My intention was to categorize the types of tests into "environment-agnostic behavior" to cover ILM's fundamental functions, and "environment-specific behavior" that differentiates between behavior specific to Cloud and that specific to on-prem. Do you think those categories make sense? Do you have any ideas on how we could name the files to reflect those intentions? Maybe `environment_agnostic_behavior` and `environment_specific_behavior`? Hmm...

Reply:
Thanks for the clarification! I hear what you're saying. To capture those three categories in this case it could be:
 
`node_allocation.helpers.ts` (for general helpers)
`cloud_environment.helpers.ts`
`on_prem_environment.helpers.ts` (not sure there are any of these though?)